faculty_cli/cli.py

```diff
@@ -473,55 +505,70 @@ def server():
-    servers = _get_servers(project_id, status=status_filter)
+    if not provided_project:
+        projects = [(project.id, project.name) for project in _list_projects()]
+        user_id = faculty_cli.auth.user_id()
+        all_servers = _get_user_servers(user_id, status=status_filter)
+        servers = defaultdict(list)
+        for server in all_servers:
+            servers[server.project_id].append(server)
+    else:
+        project_id = _resolve_project(provided_project)
+        servers = {project_id: _get_servers(project_id, status=status_filter)}
+        projects = [(project_id, "")]
 
-    if verbose:
-        if not servers:
-            click.echo("No servers.")
-        else:
-            headers = (
-                "Server Name",
-                "Type",
-                "Machine Type",
-                "CPUs",
-                "RAM",
-                "Status",
-                "ID",
-                "Started",
-            )
-            rows = []
-            for server in servers:
-                if isinstance(server.resources, SharedServerResources):
-                    machine_type = "-"
-                    cpus = "{:.3g}".format(server.resources.milli_cpus / 1000)
-                    memory_gb = "{:.3g}GB".format(
-                        server.resources.memory_mb / 1000
-                    )
-                else:
-                    machine_type = server.resources.node_type
-                    cpus = "-"
-                    memory_gb = "-"
-                rows.append(
-                    (
-                        server.name,
-                        server.type,
-                        machine_type,
-                        cpus,
-                        memory_gb,
-                        server.status.value,
-                        server.id,
-                        server.created_at.strftime("%Y-%m-%d %H:%M"),
-                    )
+    status_filter = None if all else ServerStatus.RUNNING
+    headers = (
+        "Project Name",
+        "Project ID",
+        "Server Name",
+        "Type",
+        "Machine Type",
+        "CPUs",
+        "RAM",
+        "Status",
+        "Server ID",
+        "Started",
+    )
+    found_servers = []
+    for project_id, project_name in projects:
+        for server in servers[project_id]:
+            machine_type, cpus, memory_gb = _server_spec(server)
+            found_servers.append(
+                (
+                    project_name,
+                    project_id,
+                    server.name,
+                    server.type,
+                    machine_type,
+                    cpus,
+                    memory_gb,
+                    server.status.value,
+                    server.id,
+                    server.created_at.strftime("%Y-%m-%d %H:%M"),
                 )
-            click.echo(tabulate(rows, headers, tablefmt="plain"))
+            )
+    if not found_servers:
+        print("No servers.")
```

Comment:
This was deliberately not included in the verbose case before as we wanted it to work well as a composable shell command. If you imagine someone doing:

```bash
faculty server list myproject | xargs dosomething
```

Then in the case that there are no servers, `dosomething` will be applied to "No servers.". For this reason, we prefer no output in the empty-result non-verbose case.

Does this make sense?